src/codegen.cpp

```diff
@@ -5640,6 +5625,25 @@ static void init_julia_llvm_env(Module *m)
                                      "jl_float_temp"));
 #endif
 
+#ifndef JULIA_ENABLE_THREADING
+    jltls_states_var =
+        new GlobalVariable(*m, ArrayType::get(T_int8, sizeof(jl_tls_states_t)),
```

Comment:
Done. You are right that `sizeof(jl_tls_states_t) % sizeof(void*) == 0`. The bitcast was actually not necessary anyway since it's always casted to `T_ppjlvalue` by `emit_nthptr_addr` anyway....